### PR TITLE
Addresses BACKLOG-40050: Pause and resume schedules

### PR DESCRIPTION
**Pull request by mbrasil**
@pentaho/millenniumfalcon please review

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 4s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core,ui
```
---
:ok_hand: All tests passed!
 
Tests run: **362**, Failures: **0**, Skipped: **12** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-154/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java [hunk 1]
```diff
@@ -388,7 +388,7 @@ public String shutdown() throws SchedulerException {
   public JobState pauseJob( String jobId ) throws SchedulerException {
     Job job = (Job) getJob( jobId );
 
-    if ( isScheduleAllowed() || getSession().getName().equals( job.getUserName() ) ) {
+    if ( isScheduleAllowed() || isExecuteScheduleAllowed() || getSession().getName().equals( job.getUserName() ) ) {
```
Comment (afrjorge) [inline core/src/main/java/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java:391]: missing unit tests for new condition and existing user name match

File: core/src/main/java/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java [hunk 2]
```diff
@@ -400,7 +400,7 @@ public JobState pauseJob( String jobId ) throws SchedulerException {
   public JobState resumeJob( String jobId ) throws SchedulerException {
     Job job = (Job) getJob( jobId );
 
-    if ( isScheduleAllowed() || getSession().getName().equals( job.getUserName() ) ) {
+    if ( isScheduleAllowed() || isExecuteScheduleAllowed() || getSession().getName().equals( job.getUserName() ) ) {
```
Comment (afrjorge) [inline core/src/main/java/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java:403]: missing unit tests for new condition and existing user name match

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 21s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core,ui
```
---
:ok_hand: All tests passed!
 
Tests run: **362**, Failures: **0**, Skipped: **12** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-154/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 14s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core,ui
```
---
:ok_hand: All tests passed!
 
Tests run: **364**, Failures: **0**, Skipped: **12** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-154/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by afrjorge — APPROVED** [review]

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 29s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core,ui
```
---
:ok_hand: All tests passed!
 
Tests run: **370**, Failures: **0**, Skipped: **12** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-154/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by Pancho7 — APPROVED** [review]

**Pull request merged**